bodhi/server/validators.py

```diff
@@ -989,6 +989,39 @@ def validate_comment_id(request, **kwargs):
         request.errors.status = HTTPNotFound.code
 
 
+def validate_new_comment(request, **kwargs):
+    """
+    Unflatten the content of new comment to accord with 'SaveCommentSchema' data structure.
+
+    For list of bug_feedback and testcase_feedback, the request object has data in format as:
+        "bug_feedback.0.bug_id": "1234", "bug_feedback.0.karma": "1"
+    which, in order for schema to be validated, needs to be converted as follows:
+        bug_feedback: [ { "bug_id": "1234", "karma": "1" }]
+
+    Args:
+        request (pyramid.util.Request): The current request.
+        kwargs (dict): The kwargs of the related service definition. Unused.
+    """
+    # In-method import used because there is mutual referencing between
+    # 'bodhi.server.schemas' and 'bodhi.server.validators'
+    from bodhi.server.schemas import SaveCommentSchema
+
+    content = str(request.body)
+
+    if content:
+        content = eval(content)
+
+        if type(content) is dict:
+            if 'bug_feedback' not in content and 'testcase_feedback' not in content:
+                content = SaveCommentSchema().unflatten(content)
+            content = str(content)
+
+    if type(content) is str:
+        request.body = content.replace("'", "\"")
+    if type(content) is bytes:
+        request.body = content.replace(b"'", b"\"")
```

Comment:
What is the use case of receiving bytes in the body ?

Reply:
Testing using python 3, it sets the ```request.body``` as ```bytes``` and in that case, force converting into ```str``` causes the following error:
```TypeError: You can only set Request.body to bytes (not <class 'str'>)```

That's why, I kept separate logics for python 2 and 3.